src/Monitor/Monitor/ScheduledQueryRules/NewScheduledQueryRuleCommand.cs

```diff
@@ -97,35 +97,48 @@ public class NewScheduledQueryRuleCommand : ManagementCmdletBase
         [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
         #endregion
+
         protected override void ProcessRecordInternal()
         {
             try
             {
                 // Convert Tag parameter from Hashtable to Dictionary<string, string>
                 Dictionary<string, string> tags = TagsConversionHelper.CreateTagDictionary(Tag, true);
 
-                var alertingAction = new AlertingAction(severity: Action.Severity, aznsAction: Action.AznsAction, trigger: Action.Trigger, throttlingInMin: Action.ThrottlingInMin);
+                var alertingAction = new AlertingAction(severity: Action.Severity, aznsAction: Action.AznsAction,
+                    trigger: Action.Trigger, throttlingInMin: Action.ThrottlingInMin);
 
-                var parameters = new LogSearchRuleResource(location: Location, source: Source, schedule: Schedule,
-                    action:alertingAction, tags: tags, description: Description, enabled: Enabled? "true" : "false");
+                var parameters = new LogSearchRuleResource(location: NormalizeLocation(Location), source: Source,
+                    schedule: Schedule,
+                    action: alertingAction, tags: tags, description: Description, enabled: Enabled ? "true" : "false");
 
                 parameters.Validate();
                 if (this.ShouldProcess(this.Name,
                     string.Format("Creating Log Alert Rule '{0}' in resource group {0}", this.Name,
-                    this.ResourceGroupName)))
+                        this.ResourceGroupName)))
                 {
 
                     var result = this.MonitorManagementClient.ScheduledQueryRules
-                    .CreateOrUpdateWithHttpMessagesAsync(resourceGroupName: ResourceGroupName, ruleName: Name,
-                        parameters: parameters).Result;
+                        .CreateOrUpdateWithHttpMessagesAsync(resourceGroupName: ResourceGroupName, ruleName: Name,
+                            parameters: parameters).Result;
 
                     WriteObject(new PSScheduledQueryRuleResource(result.Body));
                 }
             }
             catch (Exception ex)
             {
                 throw new Exception("Error occurred while creating Log Alert rule", ex);
-            }    
+            }
+
+        }
+
+        private string NormalizeLocation(string location)
```

Comment:
@shpimpal we have a method, [`Canonicalize()`](https://github.com/Azure/azure-powershell-common/blob/master/src/ResourceManager/Version2016_09_01/Utilities/Models/Location.cs#L38-L51), that does this type of transformation to a string (as well as strips out `-` characters) -- can we use that instead of adding this `NormalizeLocation` method to multiple files?

Reply:
@cormacpayne  I'm unable to use the `Canonicalize()` method as it is not public.